### PR TITLE
HCAP | End of ROS Notification

### PR DESCRIPTION
**Pull request by taraepp**
Ticket: https://freshworks.atlassian.net/browse/HCAP-1189

For sure some room for improvement in prettifyStatus, particularly interested in feedback on:
- change the name of the file? Took all the front-end stuff out of here, name no longer reflects purpose
- similarly, put ParticipantStatus in components/generic. _Is it_ generic? Didn't see any better fits, does it justify its own new directory?
- I sacrificed brevity for clarity and made lots of variables. Looking for opportunities to simplify this where possible while maintaining readability.
- I have a boolean `isHiredByPeer = status[1] === 'hired_by_peer'`, in another place the logic was looking for `status.includes('hired_by_peer')`- will these **always** be the same? If so I'd like to simplify it. Probably other variables/expressions that are redundant.

### CODE REVIEW

**Comment by alva-fresh** [comment]
Most of the code smells seem reasonable to me - could be a good idea to take a look?

File: client/src/utils/prettifyStatus.js [hunk 1]
```diff
@@ -1,148 +1,187 @@
...
 import { capitalizedString } from './gen-util';
+import { addYearToDate } from './date';
 
 /**
- * Returns participant stats message based on the first status provided
- *
- * @param {boolean} isMoH - is this view displayed for MoH user
- * @param {string} status - first status defined in row.status
- * @returns {string} specific status message based on value / capitalized status
+ * Data required to render the Participant Status,
```
Comment (alva-fresh) [inline client/src/utils/prettifyStatus.js:5]: Thank you for taking time to refactor this! Looks wonderful! 😍 ✨

File: client/src/components/generic/Notifications.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const useStyles = makeStyles(() => ({
+  alertWarning: {
+    fontSize: '16px',
+    color: '#6c4a00',
+    backgroundColor: '#f9f1c6',
+    border: '1px solid #faebcc',
+  },
+}));
+
+export const Notifications = ({ notifications }) => {
```
Comment (fwpushan) [inline client/src/components/generic/Notifications.js:15]: Can we make this a generic notification component?

File: client/src/components/generic/ParticipantStatus.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import InfoIcon from '@material-ui/icons/Info';
+import { ComponentTooltip } from './ComponentTooltip';
+import { Button } from './Button';
+import { getParticipantStatusData } from '../../utils/prettifyStatus';
+
+export const ParticipantStatus = ({
```
Comment (fwpushan) [inline client/src/components/generic/ParticipantStatus.js:7]: Nice

File: client/src/utils/prettifyStatus.js [hunk 4]
```diff
@@ -1,148 +1,187 @@
...
-export const prettifyStatus = (
-  status,
-  id,
-  tabValue,
-  handleEngage,
-  handleAcknowledge,
-  isMoH = false,
-  participantInfo = { statusInfos: [] }
-) => {
+export const getParticipantStatusData = (status, tabValue, isMoH, participantInfo) => {
```
Comment (fwpushan) [inline client/src/utils/prettifyStatus.js:14]: Nice

File: server/services/user.js [hunk 5]
```diff
@@ -13,6 +14,50 @@ const getUser = async (id) => {
...
+    .join({
+      participantStatus: {
+        relation: collections.PARTICIPANTS_STATUS,
+        type: 'LEFT OUTER',
+        on: {
+          participant_id: 'participant_id',
+        },
+      },
+    })
+    .find({
```
Comment (fwpushan) [inline server/services/user.js:34]:
```suggestion
    .find({
    'data.date::timestamp <': todayDate,
    'participantStatus.current': true,
    'participantStatus.status': 'hired',
    'participantStatus.data.site IN': sites
    })
```
1. No need for `and`
2. `participantStatus.data.site` should `IN` `sites`

**Review by fwpushan — CHANGES_REQUESTED** [review]
Nice job, please check some of my comments

File: server/services/user.js (hunk 5 above)
Comment (taraepp) [reply server/services/user.js:34]: Getting rid of the and!
The "IN" isn't necessary though, Massive defaults to "IN" as an operator just like equality when it's an array (https://massivejs.org/docs/criteria-objects#scalar-comparison)

File: client/src/pages/private/ParticipantTable.js [hunk 6]
```diff
@@ -432,7 +434,7 @@ const ParticipantTable = () => {
           'withdrawn',
           'archived',
           'reject_ack',
-        ].find((status) => row.status.includes(status));
+        ].find((s) => row.status.includes(s));
```
Comment (fwpushan) [inline client/src/pages/private/ParticipantTable.js:437]: why this change? I think status is a more readable variable naming.

File: server/services/user.js (hunk 5 above)
Comment (fwpushan) [reply server/services/user.js:34]: ok

**Review by fwpushan — COMMENTED** [review]
Please check my comment.

File: client/src/pages/private/ParticipantTable.js (hunk 6 above)
Comment (taraepp) [reply client/src/pages/private/ParticipantTable.js:437]: Me too. This was because of a code smell detected by SonarCloud- the variable _status_ had already been declared in the upper scope. Do you think engagedStatus is accurate?
